sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/AsyncLogPurge.scala

```diff
@@ -62,6 +70,24 @@ trait AsyncLogPurge extends Logging {
     }
   }
 
+  protected def purgeOldestAsync(plan: SparkPlan): Unit = {
+    if (purgeOldestRunning.compareAndSet(false, true)) {
+      asyncPurgeExecutorService.execute(() => {
+        try {
+          purgeOldest(plan)
+        } catch {
+          case throwable: Throwable =>
+            logError("Encountered error while performing async log purge", throwable)
+            errorNotifier.markError(throwable)
```

Comment:
Will this result in a query failure ?

Reply:
No, it doesn't result in failure, it just outputs the message to the log